### ISSUE TITLE
Improves CSP compatibility

### ISSUE THREAD

**Issue by jvoisin** [saimn/sigal#178]
Currently, it's near to impossible to use a tight [CSP]( http://content-security-policy.com/ ) with Sigal, it would be great to refactor it a bit to avoid `inline` elements.

**Comment by saimn** [comment]
Are there other things to do in addition to #179 ? (I don't know much about CSP, I should read some articles about that ...)

**Comment by jvoisin** [comment]
It would be super-great to remove `<script>` and `<style>` things.

Here is my current CSP:

```
add_header Content-Security-Policy "default-src 'self' pix.dustri.org ; img-src 'self' pix.dustri.org; script-src 'self' 'unsafe-inline' pix.dustri.org; object-src 'none'; frame-ancestors 'none'; style-src 'unsafe-inline' 'self' pix.dustri.org";
```

It would be great to be able to remove `unsafe-inline` from bot `script-src` and `style-src`; but I guess that this will be super-tricky to do because of the template system.

Anyway, thank you very much for caring about this issue :)

**Comment by saimn** [comment]
Ok, for the css it should be easy, but for the javascript it is indeed more tricky.

**Comment by jvoisin** [comment]
Can't templates be applied to javascript files too?

**Comment by saimn** [comment]
This is not supported currently, it would be nice to have this possibility though.

**Comment by jvoisin** [comment]
I guess that spliting the javascript in several files will harm a bit
the performances (this will be fixed with HTTP/2).

On the other hand, there is not much to split anyway.

But since I'm not a pro in web-magic, maybe there is a better solution
that I don't know about :)

**Comment by saimn** [comment]
Yes it is possible to make two js files, one for the show_map option and another for the 'simple' galleria config. The issue is with the call to `Galleria.ready` which depend on the show_map value, but maybe it is possible to call `Galleria.ready` multiple times.
https://github.com/saimn/sigal/blob/master/sigal/themes/galleria/templates/index.html#L303

**Comment by jvoisin** [comment]
It is possible to do so, but you could also check for the map's existence in javascript, and bind the key only it it exists.

**Comment by JulienPalard** [comment]
I'm able to run sigal with:
```
Content-Security-Policy: default-src 'none'; img-src 'self'; script-src 'self'; style-src 'self'; object-src 'none'
```

With a single modification: I just moved the inline javascript (`Galleria.configure(…`) in a javascript file.

**Comment by saimn** [comment]
@JulienPalard - But I guess without the `show_map` option ? As discussed above it should not be too hard to split this javascript in two files (to handle `show_map`). If you want to have a look :wink:
